File: 2017/day01/task.rb
#!/usr/bin/env ruby
filename = ARGV[0]

File.open(filename).each_line do |line|
  input = line.chomp.split('').map(&:to_i)

  sum = 0
  input.each_with_index do  |d, i|
    if d == input[(i + 1) % input.length]
      sum += d
    end
  end
  puts "Part 1: #{sum}"

  halfway = (input.length/2).round
  sum = 0
  input.each_with_index do |d, i|
    if d == input[(i + input.length + halfway) % input.length]
      sum +=d
    end
  end

  puts "Part 2: #{sum}"
end

File: 2017/day05/task.py
#!/usr/bin/env python3

import sys

insts = []

with open(sys.argv[1]) as input:
    for line in input:
        insts.append(int(line))


print(insts)

pc = 0
count = 0
while pc >= 0 and pc < len(insts):
    oldpc = pc
    inst = insts[pc]
    pc += inst
    if inst >= 3:
        insts[oldpc] -= 1
    else:
        insts[oldpc] += 1
    count += 1

print(count)
